mono/utils/mono-merp.c

```diff
@@ -64,6 +64,48 @@ os_version_string (void)
 	return "";
 }
 
+static gboolean
+starts_with(const char *pre, const char *str)
+{
+    return strncmp(pre, str, strlen(pre)) == 0;
+}
+
+static const char *
+macos_version_string (void)
+{
+	static const char *version_string;
+	/* macOS 10.13.6 or later */
+	if (!version_string) {
+		char s[256];
+		size_t size = sizeof (s);
+		if (sysctlbyname ("kern.osproductversion", s, &size, NULL, 0) >= 0)
+			version_string = s;
+		free(s);
```

Comment:
Should you be freeing a string from the stack?

Reply:
removed, thanks.